server/app/controllers/applicant/ApplicantInformationController.java

```diff
@@ -121,6 +121,25 @@ public CompletionStage<Result> setLangFromBrowser(Http.Request request, long app
             });
   }
 
+  /**
+   * When choosing a language from the switcher on the home page without being logged in, we simply
+   * redirect back with the appropriate locale set, without saving it to any particular user.
+   */
+  public CompletionStage<Result> setLangFromSwitcherWithoutApplicant(Http.Request request) {
```

Comment:
I was thinking that switching the language should actually create a guest user and start a session. We probably want it to switch back to english after 30 minutes of inactivity. What do you think?

Reply:
The language is stored in a separate cookie than the session:

<img width="328" alt="image" src="https://github.com/user-attachments/assets/8d0ab9f0-99da-4378-93fb-f5b86a8ff33f">